go/libraries/doltcore/sqle/database.go

```diff
@@ -314,6 +314,28 @@ func (db Database) GetTableInsensitiveAsOf(ctx *sql.Context, tableName string, a
 	}
 }
 
+// isDoltSystemTable checks if the current table is a dolt system table. If
+// using the search path, also checks that dolt is the current schema or exists
+// on the search path.
+func (db Database) isDoltSystemTable(ctx *sql.Context, tableName string) bool {
+	if doltdb.HasDoltPrefix(tableName) || !resolve.UseSearchPath || db.schemaName == "dolt" {
+		return true
+	}
+
+	if db.schemaName == "" {
+		schemasToSearch, err := resolve.SearchPath(ctx)
```

Comment:
This block is subtly wrong. If the user has a table named e.g. `log` and that schema is on the search path before the `dolt` schema, that name must resolve to their table. Should add a test for this case.

Reply:
You should probably write `resolve.SystemTable(ctx, tableName)` that handles this logic and call it here.